docs/rfc/rfc-011-delete-gas.md

```diff
@@ -0,0 +1,140 @@
+from the mempool. Overflowing MaxGas is the _only_ way that a transaction can be considered
+invalid that is not directly a result of failing the `CheckTx`. Operators, and the application,
+do not know that a transaction was removed from the mempool for this reason. A stateless check
+of this nature is exactly what `CheckTx` exists for and there is no reason for the mempool
+to keep track of this data separately.
+
+Removing Gas from Tendermint and the mempool would allow for the 'mempool' to be a better
+abstraction: any transaction that arrived at `CheckTx` and passed the check will either be
+a candidate for a later block or evicted after a TTL is reached. All other transactions are 
+completely invalid and can be discarded forever.
+
+Removing gas will not be completely straightforward. It will mean ensuring that 
+equivalent functionality can be implemented outside of the mempool using the mempool's API.
+
+## Discussion
+
+This section catalogs the functionality that will need to exist within the Tendermint
+mempool to allow Gas to be removed and replaced by application-side bookkeeping.
+
+### Requirement: Provide Mempool Tx Sorting Mechanism
+
+Gas produces a market for inclusion in a block. On many networks, a [gas fee][cosmos-sdk-fees] is
+included in pending transactions. This fee indicates how much a user is willing to
+pay per unit of execution and the fees are distributed to validators.
+
+Validators wishing to extract higher gas fees are incentivized to include transactions 
+with the highest listed gas fees into each block. This produces a naturally ordering
+of the pending transactions. Applications wishing to implement a gas mechanism need
+to be able to order the transactions in the mempool. This can trivially be accomplished 
+by sorting transactions using the `priority` field available to applications as part of
+v0.35's `ResponseCheckTx` message.
+
+### Requirement: Allow Application-Defined Block Resizing
+
+When creating a block proposal, Tendermint pulls a set of possible transactions out of
+the mempool to include in the next block. Tendermint uses MaxGas to limit the set of transactions
+it pulls out of the mempool fetching a set of transactions whose sum is less than MaxGas.
+
+By removing gas tracking from Tendermint's mempool, Tendermint will need to provide a way for
+applications to determine an acceptable set of transactions to include in the block.
+
+This is what the new ABCI++ `PrepareProposal` method is useful for. Applications
+that wish to limit the contents of a block by an application-defined limit may
+do so by removing transactions from the proposal it is passed during `PrepareProposal`.
+
+Fully meeting the requirements of the current Gas system would mean adding functionality that is
+not currently provided the `PrepareProposal` method.
+Applications that wish to include _more_ transactions in the proposal than `PrepareProposal`
```

Comment:
Not sure how ABCI++ changes this but iiuc, currently, `PrepareProposal` takes the tx-es returned by `ReapMaxBytesMaxGas()` which can take unspecified max bytes and gas parameters. In this case all Tx-es will be returned.  The app could then remove as it wishes. The `more` API extension also works but I wonder if in a greedy implementation in the end the app will get all chunks anyway.